Modules/StudyProgramme/classes/class.ilPrgInvalidateExpiredProgressesCronJob.php

```diff
@@ -5,122 +5,128 @@
-     * Run job
-     *
-     * @return ilCronJobResult
-     */
-    public function run()
-    {
-        $result = new ilCronJobResult();
-        foreach ($this->user_progress_db->getExpiredSuccessfulInstances() as $progress) {
-            try {
-                $progress->invalidate();
-            } catch (ilException $e) {
-                $this->log->write('an error occured: ' . $e->getMessage());
-            }
-        }
-        $result->setStatus(ilCronJobResult::STATUS_OK);
-        return $result;
-    }
-}
+/**
+ This ivalidates a successful progress if validityOfQualification is reached.
+
+ This is deprecated, I think.
+ It is perfectly feasible to raise some event for this, though,
+ but invalidation is reached by a date rather than a flag set by a cron job.
+  */
+ class ilPrgInvalidateExpiredProgressesCronJob extends ilCronJob
+ {
+     const ID = 'prg_invalidate_expired_progresses';
+
+     /**
+      * @var ilStudyProgrammeProgressRepository
+      */
+     protected $user_progress_db;
+
+     /**
+      * @var ilLog
+      */
+     protected $log;
+
+     /**
+      * @var ilLanguage
+      */
+     protected $lng;
+
+     public function __construct()
+     {
+         global $DIC;
+
+         $this->user_progress_db = ilStudyProgrammeDIC::dic()['model.Progress.ilStudyProgrammeProgressRepository'];
+         $this->log = $DIC['ilLog'];
+         $this->lng = $DIC['lng'];
+         $this->lng->loadLanguageModule('prg');
+     }
+
+     /**
+      * Get title
+      *
+      * @return string
+      */
+     public function getTitle()
+     {
+         return $this->lng->txt('prg_invalidate_expired_progresses_title');
+     }
+
+     /**
+      * Get description
+      *
+      * @return string
+      */
+     public function getDescription()
+     {
+         return $this->lng->txt('prg_invalidate_expired_progresses_desc');
+     }
+
+     /**
+      * Get id
+      *
+      * @return string
+      */
+     public function getId()
+     {
+         return self::ID;
+     }
+
+     /**
+      * Is to be activated on "installation"
+      *
+      * @return boolean
+      */
+     public function hasAutoActivation()
+     {
+         return true;
+     }
+
+     /**
+      * Can the schedule be configured?
+      *
+      * @return boolean
+      */
+     public function hasFlexibleSchedule()
+     {
+         return true;
+     }
+
+     /**
+      * Get schedule type
+      *
+      * @return int
+      */
+     public function getDefaultScheduleType()
+     {
+         return self::SCHEDULE_TYPE_IN_DAYS;
+     }
+
+     /**
+      * Get schedule value
+      *
+      * @return int|array
+      */
+     public function getDefaultScheduleValue()
+     {
+         return 1;
+     }
+
+     /**
+      * Run job
+      *
+      * @return ilCronJobResult
+      */
+     public function run()
+     {
+         $result = new ilCronJobResult();
+         foreach ($this->user_progress_db->readExpiredSuccessfull() as $progress) {
```

Comment:
```suggestion
         foreach ($this->user_progress_db->getExpiredSuccessfull() as $progress) {
```

would be the proper naming for the method imo. I mean, no matter if they are read, already in the memory or static, we just want HAVE these progresses =)